src/test/java/org/imsglobal/caliper/ri/events/GradeEventGradedTest.java

```diff
@@ -0,0 +1,152 @@
+import org.imsglobal.caliper.entities.agent.SoftwareApplication;
+import org.imsglobal.caliper.entities.agent.Status;
+import org.imsglobal.caliper.entities.outcome.Score;
+import org.imsglobal.caliper.entities.resource.Assessment;
+import org.imsglobal.caliper.entities.resource.Attempt;
+import org.imsglobal.caliper.events.GradeEvent;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+@Category(org.imsglobal.caliper.UnitTest.class)
+public class GradeEventGradedTest {
+    private JsonldContext context;
+    private String id;
+    private SoftwareApplication actor, edApp;
+    private Person learner;
+    private Attempt object;
+    private Assessment assignable;
+    private Score generated;
+    private Membership membership;
+    private GradeEvent event;
+
+    private static final String BASE_URN = "urn:uuid:";
+    private static final String BASE_IRI = "https://www.hmhco.com/";
+    private static final String APP_NAME = "ReadingInventory";
+
+
+    @Before
+    public void setUp() throws Exception {
+        context = JsonldStringContext.getDefault();
+
+        id = "urn:uuid:a50ca17f-5971-47bb-8fca-4e6e6879001d";
+
+        actor = SoftwareApplication.builder().id(BASE_IRI.concat(APP_NAME)).version("v2").build();
+        learner = Person.builder().id(BASE_URN.concat("0f4fedbe-2227-415f-8553-40731a627171"))
+                    .name("Casandra Rath").build();
+        assignable = Assessment.builder().id(BASE_URN.concat("c050e852-5edb-4743-92d1-b53466de3a5f")).build();
+
+        object = Attempt.builder()
+            .id(BASE_URN.concat("c35603f7a3434e2cb74ef14d60e70f42"))
+            .assignable(assignable)
+            .assignee(learner)
+            .count(100)
+            .dateCreated(new DateTime(2016, 11, 15, 10, 5, 0, 0, DateTimeZone.UTC))
+            .startedAtTime(new DateTime(2016, 11, 15, 10, 5, 0, 0, DateTimeZone.UTC))
+            .endedAtTime(new DateTime(2016, 11, 15, 10, 55, 12, 0, DateTimeZone.UTC))
+            .build();
+
+        generated = Score.builder()
+            .id(BASE_URN.concat("c050e852-5edb-4743-92d1-b53466de3a5f"))
+            .attempt(Attempt.builder().id(object.getId()).coercedToId(true).build())
+            .maxScore(1700)
+            .scoreGiven(800)
+            .scoredBy(SoftwareApplication.builder().id(BASE_IRI.concat(APP_NAME)).coercedToId(true).build())
+            .comment("The lexile score from Test Quiz # c35603f7a3434e2cb74ef14d60e70f42 taken on date/time for student 0f4fedbe-2227-415f-8553-40731a627171")
+            .dateCreated(new DateTime(2016, 11, 15, 10, 56, 0, 0, DateTimeZone.UTC))
+            .build();
+
+        edApp = SoftwareApplication.builder().id(BASE_IRI.concat(APP_NAME)).version("v2").build();
```

Comment:
v1 or v2 ?

Reply:
i just used v2 since thats what the default fixtures were using.

Reply:
ok, thats just sample data - that version is optional and represents the current form or version of whatever software application it represents